astrodb_utils/sources.py

```diff
@@ -0,0 +1,447 @@
+        logger.debug(
+            f"{source}: Trying coord in database search around "
+            f"{location.ra.degree}, {location.dec}"
+        )
+        db_name_matches = db.query_region(
+            location, radius=radius, ra_col=ra_col_name, dec_col=dec_col_name
+        )
+
+    # If still no matches, try to get the coords from SIMBAD using source name
+    if len(db_name_matches) == 0 and use_simbad:
+        simbad_skycoord = coords_from_simbad(source)
+        # Search database around that coordinate
+        if simbad_skycoord is not None:
+            radius = u.Quantity(search_radius, unit="arcsec")
+            msg2 = (
+                f"Finding sources around {simbad_skycoord} with radius {radius} "
+                f"using ra_col_name: {ra_col_name}, dec_col_name: {dec_col_name}"
+            )
+            logger.debug(msg2)
+            db_name_matches = db.query_region(
+                simbad_skycoord, radius=radius, ra_col=ra_col_name, dec_col=dec_col_name
+            )
+
+    if len(db_name_matches) == 1:
+        db_names = db_name_matches["source"].tolist()
+        logger.debug(f"One match found for {source}: {db_names[0]}")
+    elif len(db_name_matches) > 1:
+        db_names = db_name_matches["source"].tolist()
+        logger.debug(f"More than one match found for {source}: {db_names}")
+        # TODO: Find way for user to choose correct match
+    elif len(db_name_matches) == 0:
+        db_names = []
+        logger.debug(f" {source}: No match found")
+    else:
+        raise AstroDBError(f"Unexpected condition searching for {source}")
+
+    return db_names
+
+
+def coords_from_simbad(source):
+    """
+    Get coordinates from SIMBAD using a source name
+
+    Parameters
+    ----------
+    source: str
+        Name of the source
+
+    Returns
+    ------- 
+    SkyCoord object
+
+    """
+    simbad_result_table = Simbad.query_object(source)   
+    if simbad_result_table is None:
+        logger.debug(f"SIMBAD returned no results for {source}")
+        simbad_skycoord = None
+    elif len(simbad_result_table) == 1:
+        logger.debug(f"simbad colnames: {simbad_result_table.colnames} \n simbad results \n {simbad_result_table}")
+        if 'RA' in simbad_result_table.colnames: # for astroquery<0.4.7
+            ra_col_name_simbad = 'RA'
+            dec_col_name_simbad = 'DEC'
+        elif 'ra' in simbad_result_table.colnames: # for astroquery >=0.4.8
+            ra_col_name_simbad = 'ra'
+            dec_col_name_simbad = 'dec'
+        simbad_coords = f"{simbad_result_table[ra_col_name_simbad][0]} {simbad_result_table[dec_col_name_simbad][0]}"
+        logger.debug(f"SIMBAD coord string: {simbad_coords}")
+        simbad_skycoord = SkyCoord(simbad_coords, unit=(u.deg, u.deg))
+        ra = simbad_skycoord.to_string(style="decimal").split()[0]
+        ra = simbad_skycoord.ra.deg
+        dec = simbad_skycoord.to_string(style="decimal").split()[1]
```

Comment:
Curious as to this. Why not just have `ra = simbad_skycoord.ra.deg` and `dec = simbad_skycoord.dec.deg`? The first `ra` settings is overwritten and the dec one appears to do some string splitting instead of just fetching the decimal degree value.